pkg/bpf/bpf_linux.go

```diff
@@ -113,7 +115,29 @@ func UpdateElementFromPointers(fd int, structPtr unsafe.Pointer, sizeOfStruct ui
 	}
 
 	if ret != 0 || err != 0 {
-		return fmt.Errorf("Unable to update element for map with file descriptor %d: %s", fd, err)
+
+		var errMsg error
+
+		switch err {
+		case unix.E2BIG:
+			if strings.Contains(mapName, "lb") {
+				errMsg = errors.New("the bpf LB map is full, please consider resizing it with the flag \"bpf-lb-max-entries\". THIS WILL INTERFERE WITH ALL EXISTING CONNECTIONS")
+			} else {
+				errMsg = errors.New("the bpf map is full, please consider resizing it")
+			}
+		case unix.EEXIST:
+			errMsg = errors.New("Cannot add the element to bpf map, please check the key of the element")
+		case unix.ENOENT:
+			errMsg = errors.New("Cannot add/update element to bpf map, the element doesn't exist")
+		default:
+			errMsg = nil
+		}
+
+		if errMsg != nil {
```

Comment:
I think we can simplify the code by just returning what we want at each 'case` branch:

For example:
```
case unix.ENOENT:
     return fmt.Errorf("failed to update bpf map %s (fd: %d): key does not exist", mapName, fd)
case ...
     ...
default:
      return fmt.Errorf("failed to update bpf map %s (fd: %d): %w", mapName, fd, err)